src/main/java/seedu/address/model/person/Person.java

```diff
@@ -53,30 +59,30 @@ public Address getAddress() {
     }
 
     /**
-     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
-     * if modification is attempted.
+     * Returns an immutable tag set, which throws
+     * {@code UnsupportedOperationException} if modification is attempted.
      */
     public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
     /**
-     * Returns true if both persons of the same name have at least one other identity field that is the same.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if both persons of the same name have at least one other
+     * identity field that is the same. This defines a weaker notion of equality
+     * between two persons.
      */
     public boolean isSamePerson(Person otherPerson) {
         if (otherPerson == this) {
             return true;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName())
+        return otherPerson != null && otherPerson.getNric().equals(getNric()) && otherPerson.getName().equals(getName())
                 && (otherPerson.getPhone().equals(getPhone()) || otherPerson.getEmail().equals(getEmail()));
```

Comment:
/nit Shall we put each `&&` condition on a separate line for clarity?

Reply:
Personally I don't think it's an issue, but I can see why you'd want it. Team decision?